tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/BaseLinker.scala

```diff
@@ -434,4 +437,32 @@ final class BaseLinker(semantics: Semantics, esLevel: ESLevel, considerPositions
           s"Cannot find $methodName in ${classInfo.encodedName}")
     }
   }
+
+  private def checkConflictingExports(unit: LinkingUnit, logger: Logger,
+      bypassLinkingErrors: Boolean): Unit = {
+    val namesAndClasses = for {
+      classDef <- unit.classDefs
+      name <- classDef.exportNames
+    } yield {
+      name -> classDef
+    }
+
+    val level = if (bypassLinkingErrors) Level.Warn else Level.Error
+    val errors = for {
+      (name, namesAndClasses) <- namesAndClasses.groupBy(_._1)
+      if namesAndClasses.size > 1
+    } yield {
+      logger.log(level, s"Conflicting top-level exports to $name:")
+      logger.log(level, s"The following classes/objects all export to $name:")
```

Comment:
Those two lines of error message that repeat the essential information `$name` might confuse users as two separate errors. Consider simply adding "from the following classes:" at the end of the first message.

Not sure about this.